packages/block-library/src/columns/index.js

```diff
@@ -88,75 +77,18 @@ export const settings = {
 		html: false,
 	},
 
-	deprecated: [
-		{
-			attributes: {
-				columns: {
-					type: 'number',
-					default: 2,
-				},
-			},
-			isEligible( attributes, innerBlocks ) {
-				// Since isEligible is called on every valid instance of the
-				// Columns block and a deprecation is the unlikely case due to
-				// its subsequent migration, optimize for the `false` condition
-				// by performing a naive, inaccurate pass at inner blocks.
-				const isFastPassEligible = innerBlocks.some( ( innerBlock ) => (
-					/layout-column-\d+/.test( innerBlock.originalContent )
-				) );
-
-				if ( ! isFastPassEligible ) {
-					return false;
-				}
-
-				// Only if the fast pass is considered eligible is the more
-				// accurate, durable, slower condition performed.
-				return innerBlocks.some( ( innerBlock ) => (
-					getDeprecatedLayoutColumn( innerBlock.originalContent ) !== undefined
-				) );
-			},
-			migrate( attributes, innerBlocks ) {
-				const columns = innerBlocks.reduce( ( result, innerBlock ) => {
-					const { originalContent } = innerBlock;
-
-					let columnIndex = getDeprecatedLayoutColumn( originalContent );
-					if ( columnIndex === undefined ) {
-						columnIndex = 0;
-					}
-
-					if ( ! result[ columnIndex ] ) {
-						result[ columnIndex ] = [];
-					}
-
-					result[ columnIndex ].push( innerBlock );
-
-					return result;
-				}, [] );
-
-				const migratedInnerBlocks = columns.map( ( columnBlocks ) => (
-					createBlock( 'core/column', {}, columnBlocks )
-				) );
-
-				return [
-					attributes,
-					migratedInnerBlocks,
-				];
-			},
-			save( { attributes } ) {
-				const { columns } = attributes;
-
-				return (
-					<div className={ `has-${ columns }-columns` }>
-						<InnerBlocks.Content />
-					</div>
-				);
-			},
-		},
-	],
+	deprecated,
 
 	edit( { attributes, setAttributes, className } ) {
-		const { columns } = attributes;
-		const classes = classnames( className, `has-${ columns }-columns` );
+		const { columns, verticalAlignment } = attributes;
+		const classes = classnames( className, {
+			[ `has-${ columns }-columns` ]: true,
+			[ `are-vertically-aligned-${ verticalAlignment }` ]: ! isNil( verticalAlignment ),
```

Comment:
Is there a falsey value that we would want to consider here other than those checked by `isNil`? As in, do we want cases like:
```js
`are-vertically-aligned-${ false }` ]: ! isNil( false ), // true
`are-vertically-aligned-${ '' }` ]: ! isNil( '' ), // true
`are-vertically-aligned-${ 0 }` ]: ! isNil( 0 ), // true
```

If not, maybe 
```js
[ ... ]: ! isNil( verticalAlignment )
``` 
could just be replaced with 
```js 
[ ... ]: !! verticalAlignment
```

Reply:
I'd rather catch as many cases as possible. 

So as I understand it all the cases we're considering are [`falsey` values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), but we explicitly coerce to a boolean via the double `!!`?

Reply:
Yeah, we don't strictly need to coerce for `classnames` though as it'll apply the class with anything truthy, so we could actually skip the `!!`.

Reply:
Fixed. Thank you for this.